### PR TITLE
feat: add @assert length check into setmarginals! and setmessages!

### PR DESCRIPTION
**Pull request by Nimrais**
This PR aims to resolve https://github.com/biaslab/RxInfer.jl/issues/175.

### CODE REVIEW

File: src/variables/variable.jl [hunk 1]
```diff
@@ -93,13 +93,29 @@ getmarginals(variables::AbstractArray{<:AbstractVariable}, skip_strategy::Margin
...
+
+Base.length(ri::RepeatedValueContainer) = ri._length
+Base.iterate(ri::RepeatedValueContainer) = iterate(ri.container)
+Base.iterate(ri::RepeatedValueContainer, state) = iterate(ri.container, state)
+
+function RepeatedValueContainer(marginal, _length)
+    container = Iterators.repeated(marginal, _length)
+    return RepeatedValueContainer(marginal, container, _length)
+end
+
```
Comment (bvdmitri) [inline src/variables/variable.jl:110]:
```suggestion
```
Why is this needed? (+ it is slower than just using `Iterators.repeated` because all types are `Any`)

File: src/variables/variable.jl [hunk 2]
```diff
@@ -93,13 +93,29 @@ getmarginals(variables::AbstractArray{<:AbstractVariable}, skip_strategy::Margin
...
 setmarginal!(variable::AbstractVariable, marginal) = setmarginal!(getmarginal(variable, IncludeAll()), marginal)
 
-setmarginals!(variables::AbstractArray{<:AbstractVariable}, marginal::PointMass)    = _setmarginals!(Base.HasLength(), variables, Iterators.repeated(marginal, length(variables)))
-setmarginals!(variables::AbstractArray{<:AbstractVariable}, marginal::Distribution) = _setmarginals!(Base.HasLength(), variables, Iterators.repeated(marginal, length(variables)))
+setmarginals!(variables::AbstractArray{<:AbstractVariable}, marginal::PointMass)    = _setmarginals!(Base.HasLength(), variables, RepeatedValueContainer(marginal, length(variables)))
+setmarginals!(variables::AbstractArray{<:AbstractVariable}, marginal::Distribution) = _setmarginals!(Base.HasLength(), variables, RepeatedValueContainer(marginal, length(variables)))
```
Comment (bvdmitri) [inline src/variables/variable.jl:114]:
```suggestion
setmarginals!(variables::AbstractArray{<:AbstractVariable}, marginal::PointMass)    = _setmarginals!(Base.HasLength(), variables, Iterators.repeated(marginal, length(variables)))
setmarginals!(variables::AbstractArray{<:AbstractVariable}, marginal::Distribution) = _setmarginals!(Base.HasLength(), variables, Iterators.repeated(marginal, length(variables)))
```

File: src/variables/variable.jl (hunk 1 above)
Comment (Nimrais) [reply src/variables/variable.jl:110]: @bvdmitri you are right It's not needed, my reasoning about iterators length function was wrong. I checked the docs.

File: src/variables/variable.jl [hunk 3]
```diff
@@ -114,11 +130,12 @@ end
 setmessage!(variable::AbstractVariable, index::Int, message) = setmessage!(messageout(variable, index), message)
 setmessage!(variable::AbstractVariable, message)             = foreach(i -> setmessage!(variable, i, message), 1:degree(variable))
 
-setmessages!(variables::AbstractArray{<:AbstractVariable}, message::PointMass)    = _setmessages!(Base.HasLength(), variables, Iterators.repeated(message, length(variables)))
-setmessages!(variables::AbstractArray{<:AbstractVariable}, message::Distribution) = _setmessages!(Base.HasLength(), variables, Iterators.repeated(message, length(variables)))
+setmessages!(variables::AbstractArray{<:AbstractVariable}, message::PointMass)    = _setmessages!(Base.HasLength(), variables, RepeatedValueContainer(message, length(variables)))
+setmessages!(variables::AbstractArray{<:AbstractVariable}, message::Distribution) = _setmessages!(Base.HasLength(), variables, RepeatedValueContainer(message, length(variables)))
```
Comment (bvdmitri) [inline src/variables/variable.jl:134]:
```suggestion
setmessages!(variables::AbstractArray{<:AbstractVariable}, message::PointMass)    = _setmessages!(Base.HasLength(), variables, Iterators.repeated(message, length(variables)))
setmessages!(variables::AbstractArray{<:AbstractVariable}, message::Distribution) = _setmessages!(Base.HasLength(), variables, Iterators.repeated(message, length(variables)))
```

File: test/variables/test_variable.jl [hunk 4]
```diff
@@ -39,6 +39,7 @@ using Test, ReactiveMP, Rocket, BayesBase, Distributions, ExponentialFamily
             activate!.(variablesmv, TestOptions())
 
             setmarginals!(variablesmv, dist)
+            @test_throws AssertionError setmarginals!(variablesmv, [dist])
```
Comment (bvdmitri) [inline test/variables/test_variable.jl:42]:
```suggestion
            @test_throws AssertionError setmarginals!(variablesmv, dist[begin:end-1])
```

File: test/variables/test_variable.jl [hunk 5]
```diff
@@ -61,6 +62,7 @@ using Test, ReactiveMP, Rocket, BayesBase, Distributions, ExponentialFamily
             activate!.(variablesmx, TestOptions())
 
             setmarginals!(variablesmx, dist)
+            @test_throws AssertionError setmarginals!(variablesmx, [dist])
```
[the remark this answers is not in the document]
Comment (Nimrais) [reply test/variables/test_variable.jl]: I missed this one. What do you mean?

[the remark this answers is not in the document]
Comment (bvdmitri) [reply test/variables/test_variable.jl]: @Nimrais 
I just noticed that the purpose of this test might be a bit unclear (in the future), as it currently seems to throw errors for two reasons rather than solely due to a mismatched length. Specifically, since `dist` is a matrix and `[dist]` is an array of matrices, this combination isn't supported and may also throw. To effectively test the `length`, you can exclude the last element from `dist`. A straightforward way to achieve this is by using `dist[begin:end-1]`, which will include all elements of `dist` except the last one and ensure that the length is equal to `N - 1`.

[the remark this answers is not in the document]
Comment (Nimrais) [reply test/variables/test_variable.jl]: We do not have any multivariate distributions in this test, only this ones: NormalMeanVariance(-2.0, 3.0), NormalMeanPrecision(-2.0, 3.0), PointMass(2.0).

[the remark this answers is not in the document]
Comment (bvdmitri) [reply test/variables/test_variable.jl]: Ah sorry, you are right, I confused myself with the `variablesmx` variable. The updated tests are also good!

[the remark this answers is not in the document]
Comment (Nimrais) [reply test/variables/test_variable.jl]: But I will add it now, your concern is reasonable.

[the remark this answers is not in the document]
Comment (bvdmitri) [reply test/variables/test_variable.jl]: As I can see you've added it [here](https://github.com/biaslab/ReactiveMP.jl/pull/366/commits/de20edac1bceeb9b72d546e7c796ec56f6013e1f) already. This is enough :)

**Review by bvdmitri — APPROVED** [review]
